src/hardware/setup.rs

```diff
@@ -870,22 +870,35 @@ pub fn setup(
             };
 
             #[cfg(not(feature = "pounder_v1_0"))]
-            let reset_pin = gpiog.pg6.into_push_pull_output();
+            let mut reset_pin = gpiog.pg6.into_push_pull_output();
             #[cfg(feature = "pounder_v1_0")]
-            let reset_pin = gpioa.pa0.into_push_pull_output();
+            let mut reset_pin = gpioa.pa0.into_push_pull_output();
 
             let mut io_update = gpiog.pg7.into_push_pull_output();
 
             let mut ad9959 = ad9959::Ad9959::new(
                 qspi_interface,
-                reset_pin,
-                &mut io_update,
-                &mut delay,
-                ad9959::Mode::FourBitSerial,
                 design_parameters::DDS_REF_CLK.to_Hz() as f32,
-                design_parameters::DDS_MULTIPLIER,
-            )
-            .unwrap();
+            );
+
+            // The oscillator to the AD9959 takes some time to turn on after first initialization.
+            // To accomodate this, we retry the initialization process until the configuration
+            // process completes successfully.
+            loop {
```

Comment:
Nono. This would potentially wait forever.
Also it's not a good idea to do the configuration when the clock is just coming up and leave it there. There is no certain indicator that the clock is fine and the initialization is fine under those conditions.
Just a sufficient delay (take 10 ms) is the right thing to do.

Reply:
Also we should explicitly `set_ext_clk(false)` once very early and wait after that for oscillator power up.

Reply:
>Just a sufficient delay (take 10 ms) is the right thing to do.

The main problem I have is that the datasheet for the internal clock does not specify a startup time. We have a watchdog going for 30s during initialization that would catch anything infinite. That being said, I have nothing opposed to inserting a predefined delay and calling it good. It would just be nicer to have justification for said delay.

> Also we should explicitly `set_ext_clk(false)` once very early and wait after that for oscillator power up.

This is done during `PounderDevices::new()` above, so we don't have to worry about it here.

Reply:
The DDS PLL needs ~1 ms to lock. But that's not critical here since at this point we don't use it yet (but we should add a 1 ms delay after `configure_system_clock()` at this point). The XO has ~ 2 ms startup time. That's what we are waiting for. We do need to worry about that.

Reply:
Where are you getting the ~2ms startup time for the XO from? I don't see anything in the [datasheet](https://eu.mouser.com/datasheet/2/94/CCHD_950-2998865.pdf)

Reply:
Extrapolation from [others of the same architecture/style](https://www.crystek.com/crystal/spec-sheets/clock/cchd-957.pdf) that have it spec'd.